components/motor/gpiostepper/gpiostepper.go

```diff
@@ -288,7 +284,9 @@ func (m *gpioStepper) doCycle(ctx context.Context) (time.Duration, error) {
 func (m *gpioStepper) doStep(ctx context.Context, forward bool) error {
 	err := multierr.Combine(
 		m.dirPin.Set(ctx, forward, nil),
-		m.stepPin.Set(ctx, true, nil))
+		m.stepPin.Set(ctx, true, nil),
+		m.enable(ctx, true),
```

Comment:
so as long as the motor thinks it should be moving (`m.stepPosition == m.targetStepPosition`), the motor will be enabled. 

seems like we might be able to put all of the m.enable calls inside of `doCycle`, which might make the code easier to follow, but def not necessary.

Reply:
Yes, also +1 on making this code easier to read and reason about, it's in a state right now. I will follow up.